sdk/rust/oak_runtime_compat/src/asylo/allocator.rs

```diff
@@ -73,3 +76,36 @@ unsafe impl GlobalAlloc for System {
         free(ptr);
     }
 }
+
+#[link(name = "sgx_trts")]
+extern "C" {
+    // SGX-enabled abort function that causes an undefined instruction (`UD2`) to be executed, which
+    // terminates the enclave execution.
+    //
+    // The C type of this function is `extern "C" void abort(void) __attribute__(__noreturn__);`
+    //
+    // See https://github.com/intel/linux-sgx/blob/d166ff0c808e2f78d37eebf1ab614d944437eea3/sdk/trts/linux/trts_pic.S#L565.
+    fn abort() -> !;
+}
+
+
+#[global_allocator]
```

Comment:
I think it's good practice to only declare the allocator struct in a crate, but expect the actual binding to happen at some higher level in the code. See https://docs.rs/wee_alloc/0.4.5/wee_alloc/ as an example.

Reply:
I've moved 
~~~rust 
#[global_allocator]
static A: asylo::allocator::System = asylo::allocator::System;
~~~
to the top level `lib.rs` - is that what you were suggesting?